### PR TITLE
[FBcode->GH] Fix some tests in fbcode

### PR DESCRIPTION
**Pull request by NicolasHug**
Cherry-picking of #3675, #3676 #3677 and https://github.com/pytorch/vision/pull/3687

### CODE REVIEW

**Review by fmassa — APPROVED** [review]
LGTM, thanks!

Please make sure that the commit message contains the `[FBcode->GH]` prefix before merging the PR

**Comment by NicolasHug** [comment]
@datumbox do you foresee any potential issue with squeezing the 3 commits into one as done here, instead of opening one PR for each of them?

**Comment by datumbox** [comment]
@NicolasHug No major problem. Squashing used to be the way we synched the repos in the past anyway. 

Ideally when possible it would be good to bring them as individual PRs. It's more of a nice to have for tracking and managing reasons. Should not block merging this one IMO. :)

**Pull request merged**
